syntax = "proto3";

package syft.core.node.domain.service;

import "proto/core/common/common_object.proto";
import "proto/core/io/address.proto";
import "proto/lib/python/dict.proto";

message UpdateRequestHandlerMessage {
  syft.core.io.Address address = 1;
  syft.lib.python.Dict handler = 2;
  bool keep = 4;
}

message GetAllRequestHandlersMessage {
  syft.core.io.Address address = 1;
  syft.core.io.Address reply_to = 2;
}

message GetAllRequestHandlersResponseMessage {
  syft.core.io.Address address = 1;
  repeated syft.lib.python.Dict handlers = 2;
}
